src/dune_rules/melange/melange_rules.ml

```diff
@@ -307,12 +307,13 @@ let setup_js_rules_libraries ~dir ~scope ~target_dir ~sctx ~requires_link ~mode
       let* source_modules = impl_only_modules_defined_in_this_lib sctx lib in
       Memo.parallel_iter source_modules ~f:(build_js ~dir ~output ~includes))
 
+let emit_target_dir (emit : Melange_stanzas.Emit.t) ~dir =
```

Comment:
what do you think about moving the `for_melange` type in `gen_rules` to this module instead? then this function could take that single argument instead of two.

Reply:
I'm not sure it's worth it. The only point of the record is to make sure the stanza directory isn't confused with the rule directory. Making this type public wouldn't really improve the API in Melange_rules.

Reply:
Fair